routing/datasource.go

```diff
@@ -486,7 +487,7 @@ func processRouteDef(cpm map[string]PredicateSpec, fr filters.Registry, def *esk
 		return nil, err
 	}
 
-	r := &Route{Route: *def, Scheme: scheme, Host: host, Predicates: cps, Filters: fs, weight: weight}
+	r := &Route{Route: *def, Scheme: scheme, Host: host, Predicates: cps, Filters: fs, weight: weight, Rnd: rand.New(NewLockedSource())} // #nosec
```

Comment:
Is it okay to create per-route random numbers generator? Maybe there is better place?

Reply:
Do we need random per route?

Reply:
I need some random number generator to calculate fadeIn threshold for every request here 
https://github.com/zalando/skipper/pull/2634/files#diff-1d74870266949f1927a500e5f0002617239d3e3cd9bcceee6be014738e5e0f2bR484

It is not obvious for me where this random number generator should come from. Putting it in the `routing.Route` seems the least bad decision for this, but I still think there's some room for improvement, that's why I asked this question.

Reply:
What could be a problem of having one per route?

Reply:
Route is an exported interface. In go 1.21 we do not need custom random at all and can use global (see https://github.com/zalando/skipper/issues/2433)